jaggery-modules/rxt/module/scripts/asset/asset.js

```diff
@@ -369,25 +371,27 @@ var asset = {};
      * @param  {Object} options A JSON object of the asset instance to be updated
      */
     AssetManager.prototype.update = function(options) {
-        metrics.start(this.constructor.name,'update');
-        var isDefault = false;
-        if ((options.hasOwnProperty(constants.Q_PROP_DEFAULT)) && (options[constants.Q_PROP_DEFAULT] === true)) {
-            isDefault = true;
-        }
-        if(options.attributes.hasOwnProperty(constants.ASSET_PROVIDER)){
-            options.attributes[constants.ASSET_PROVIDER] = options.attributes[constants.ASSET_PROVIDER].replace('@', ':');
-        }
-        this.am.update(options);
-        var asset = this.am.get(options.id);
-        if (!this.rxtManager.isGroupingEnabled(this.type)) {
-            log.debug('Omitting grouping step as the groupingEnabled property in the asset configuration has been disabled');
+        try {
+            metrics.start(this.constructor.name,'update');
+            var isDefault = false;
+            if ((options.hasOwnProperty(constants.Q_PROP_DEFAULT)) && (options[constants.Q_PROP_DEFAULT] === true)) {
+                isDefault = true;
+            }
+            if(options.attributes.hasOwnProperty(constants.ASSET_PROVIDER)){
```

Comment:
Thanks @splinter , I have fixed it in [this commit](https://github.com/tmkasun/carbon-store/commit/a5917ea67af4d60979a9f5f59636c7672843645b)